java/registry/src/main/java/dev/sunbirdrc/registry/dao/VertexReader.java

```diff
@@ -485,14 +485,40 @@ public JsonNode readInternal(Vertex rootVertex) throws Exception {
         // objects.
         // The properties could exist anywhere. Refer to the local arrMap.
         expandChildObject(rootNode, 0);
-
+        if(expandReferenceObj)
+            expandReferenceNodes(rootNode);
         entityNode.set(entityType, rootNode);
 
+
         // After reading the entire type, now trim the @type property
         trimAttributes(entityNode);
         return entityNode;
     }
 
+    private void expandReferenceNodes(ObjectNode rootNode) {
+        rootNode.fields().forEachRemaining(entry -> {
+            if(entry.getValue().isValueNode() && entry.getValue().asText().startsWith(DID_TYPE)) {
+                String[] dids = entry.getValue().asText().split(":");
+                String osid = RecordIdentifier.parse(dids[2]).getUuid();
```

Comment:
This should throw index not found exception when it's not having size 3 or say it's not a did but starts with word did, so instead of checking for starts with `DID_TYPE` in the above condition, we can check for a pattern which should match all the Dids only i.e. _did:.+:.+_

Reply:
Added check based on the regex for DID format.